ext4/tar2ext4/tar2ext4.go

```diff
@@ -85,23 +86,29 @@ func ConvertTarToExt4(r io.Reader, w io.ReadWriteSeeker, options ...Option) erro
 			return err
 		}
 
-		if err = fs.MakeParents(hdr.Name); err != nil {
-			return errors.Wrapf(err, "failed to ensure parent directories for %s", hdr.Name)
+		// compactext assumes all paths are `/` separated
+		// [filepath.ToSlash] is platform dependent; so will be a nop on Linux, but replace
+		// '\' with '/' on Windows.
+		name := filepath.ToSlash(hdr.Name)
+		linkName := filepath.ToSlash(hdr.Linkname)
+
+		if err = fs.MakeParents(name); err != nil {
+			return errors.Wrapf(err, "failed to ensure parent directories for %s", name)
 		}
 
 		if p.convertWhiteout {
-			dir, name := path.Split(hdr.Name)
+			dir, name := path.Split(name)
```

Comment:
Could we use different name for this `name` variable (or maybe for the variable declared on line 92)? This can be easily missed.